docs/02-app/01-building-your-application/06-optimizing/02-videos.mdx

```diff
@@ -0,0 +1,281 @@
+
+   ```jsx filename="app/page.jsx"
+   import { Suspense } from 'react'
+   import { VideoComponent } from '../ui/VideoComponent.server'
+
+   export default function Page() {
+     return (
+       <section>
+         <Suspense fallback={<p>Loading video...</p>}>
+           <VideoComponent />
+         </Suspense>
+         {/* Other content of the page */}
+       </section>
+     )
+   }
+   ```
+
+> **Good to know**: When embedding videos from external platforms, consider the following best practices:
+>
+> - **Responsive Embedding**: Ensure the video embeds are responsive. Use CSS to make the iframe or video player adapt to different screen sizes.
+> - **Consider Data Usage**: Implement strategies for loading videos based on network conditions, especially for users with limited data plans.
+> - **Accessibility Considerations**: Provide captions or alternative content for accessibility purposes.
+
+By integrating server components with React Suspense for loading, Next.js applications can efficiently handle externally hosted videos. This approach results in quicker load times and a smoother user experience. Such integration boosts both performance and engagement, ensuring an optimal video viewing experience on your site.
+
+## Self-Hosted Video Strategies in Next.js
+
+Self-hosting videos in Next.js may be preferable for several reasons, including control, trust, and customization. This approach is particularly beneficial when external services don't meet your specific needs or when you require complete command over video content.
+
+### Video Storage Considerations
+
+- **Complete Control and Independence**: Self-hosting gives you direct management over your video content, from playback to appearance, ensuring full ownership and control, free from external platform constraints.
+- **Customization for Specific Needs**: Ideal for unique requirements, like dynamic background videos, it allows for tailored customization to align with design and functional needs.
+- **Performance and Scalability Considerations**: Choose storage solutions that are both high-performing and scalable, to support increasing traffic and content size effectively.
+- **Cost and Integration**: Balance the costs of storage and bandwidth with the need for easy integration into your Next.js framework and broader tech ecosystem.
+
+### Example: Using Vercel Blob for Video Hosting
+
+[Vercel Blob](https://vercel.com/docs/storage/vercel-blob) offers an efficient way to self-serve videos, providing a scalable cloud storage solution that's well-integrated with Next.js. Here's how you can host a video using Vercel Blob:
+
+**Step 1: Uploading a Video to Vercel Blob**
+
+Create a server-side function to upload your video:
```

Comment:
It might be helpful to name the specific Next.js feature used instead of 'server-side function'. 

Perhaps we can align the Vercel docs terminology, and link to the two methods of uploading, [Server Uploads](https://vercel.com/docs/storage/vercel-blob/server-upload) and [Client Uploads](https://vercel.com/docs/storage/vercel-blob/client-upload). 

If you want to show an example for a server upload, you can specifically name Route Handler or Server Action.

Reply:
Don't forget to mention they need to [add the hostname to next.config.js](https://vercel.com/docs/storage/vercel-blob/server-upload#:~:text=Then%2C%20add%20the%20hostname%20to%20your%20next.config.js%20file%20including%20the%20store%20id%20from%20the%20dashboard%3A)